sstmac/common/event_scheduler.h

```diff
@@ -374,32 +374,37 @@ class EventScheduler : public sprockit::printable
 
 };
 
+namespace detail {
+  // Class to provoide the same virtual interface as SST::Subcomponent, mainly
+  // to silence warnings about override, but also because we don't change the
+  // interface when Core is unavalible. 
+  class CoreIntegratedComponentStub {
+    public:
+      virtual ~CoreIntegratedComponentStub() = default;
+      virtual void setup() = 0;
+      virtual void init(unsigned int) = 0;
+  };
+}
+
 /**
  * The interface for something that can schedule messages
  */
 class Component :
 #if SSTMAC_INTEGRATED_SST_CORE
   public SSTIntegratedComponent,
+#else
+  public detail::CoreIntegratedComponentStub,
```

Comment:
Why can't event scheduler just provide what CoreIntegratedCoponentStub does?

Reply:
I didn't see a reason to put methods into EventScheduler that have zero to do with EventScheduler.  If we ever want to refactor or change EventScheduler you would then have to try and reason about why it had these virtual methods that have nothing to do with it.

Reply:
EventScheduler is supposed to capture functionality that would be present in core. As of right now, we do not call setup or init functions when building the machine, but we might start doing that more.  Let's delete the new stub class and put the virtual methods in `EventScheduler` and be done with it. `EventScheduler` should as close as possible resemble `BaseComponent` from sst-core.